var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -67,68 +72,105 @@ def setup_dependent_package(self, module, dep_spec):
+                # installations by (unconciously) setting environment variables
+                # to switch between different interface and threading layers
+                # dynamically. From this perspective it is no different from
+                # throwing away RPATH's and using LD_LIBRARY_PATH.
+
+                # library_dirs are in compilers_and_libraries_2017/linux/mkl/lib/intel64  # noqa
+                # include_dirs are in compilers_and_libraries_2017/linux/mkl/include      # noqa
+                lapackblas_includes = set()
+                for libdir in lapackblas_libraries.directories:
+                    incdir = os.path.dirname(os.path.dirname(libdir))
+                    incdir = join_path(incdir, 'include')
+                    lapackblas_includes.add(incdir)
+
+                f.write('[mkl]\n')
+                f.write('mkl_libs = {0}\n'.format('mkl_rt'))
+                f.write('lapack_libs = \n')
+
+            elif '^atlas' in spec:
+                f.write('[atlas]\n')
+                f.write('atlas_libs = {0}\n'.format(libraries))
+
+            else:
+                # The section title for the defaults changed in @1.10
+                if spec.satisfies('@1.10:'):
+                    f.write('[ALL]\n')
                 else:
-                    # The section title for the defaults changed in @1.10, see
-                    # https://github.com/numpy/numpy/blob/master/site.cfg.example
-                    if spec.satisfies('@:1.9.2'):
-                        f.write('[DEFAULT]\n')
-                    else:
-                        f.write('[ALL]\n')
-                    f.write('libraries=%s\n'    % names)
-
-                f.write('library_dirs=%s\n' % dirs)
-                if not ((platform.system() == "Darwin") and
-                        (platform.mac_ver()[0] == '10.12')):
-                    f.write('rpath=%s\n' % dirs)
+                    f.write('[DEFAULT]\n')
+                f.write('libraries = {0}\n'.format(libraries))
+
+            include_dirs = os.pathsep.join(lapackblas_includes)
+            library_dirs = os.pathsep.join(lapackblas_libraries.directories)
+
+            f.write('include_dirs = {0}\n'.format(include_dirs))
+            f.write('library_dirs = {0}\n'.format(library_dirs))
+
+            # macOS 10.12 does not support RPATHs
+            if not ((platform.system() == "Darwin") and
+                    (platform.mac_ver()[0] == '10.12')):
+                f.write('rpath = {0}\n'.format(library_dirs))
 
     def build_args(self, spec, prefix):
         args = []
 
         # From NumPy 1.10.0 on it's possible to do a parallel build
         if self.version >= Version('1.10.0'):
-            args = ['-j', str(make_jobs)]
+            args.extend(['-j', str(make_jobs)])
+
+        if self.compiler.name == 'intel':
+            if spec.satisfies('target=x86_64'):
+                args.extend(['--compiler=intelem', '--fcompiler=intelem'])
```

Comment:
Whether or not I add these flags seems to make a big difference. If I don't supply them, it still uses `icc`, but I think it is passing GCC flags. If I use these flags, the build crashes when it can't find `xiar` because it isn't in my PATH.